pkg/async/async.go

```diff
@@ -114,6 +121,42 @@ func RunBackground(ctx context.Context, r Runner) {
 	Run(context.Background(), r)
 }
 
+// RunClose closes any references a runner might be using
+func RunClose(ctx context.Context, r Runner) error {
+	switch r := r.(type) {
+	case Closer:
+		return r.Close(ctx)
+	case io.Closer:
+		return r.Close()
+	}
+	return nil
+}
+
+// RunGroup runs a group of runner tasks and exits when the first run group errors out
+func RunGroup(rg []Runner) Runner {
+	ru := Func(func(ctx context.Context) error {
+		g, ctx := errgroup.WithContext(ctx)
+		for idx := range rg {
+			r := rg[idx]
+			g.Go(func() error {
+				defer func() {
+					if err := RunClose(ctx, r); err != nil {
+						log.Error(ctx, "Error when closing:", events.NewErrorInfo(err))
+					}
+				}()
+
+				err := r.Run(ctx)
+				if err != nil {
+					<-ctx.Done()
+				}
+				return err
```

Comment:
Why block here until context is canceled?  It will not allow an individual activity to error out and cancel the other activities?

```suggestion
				return r.Run(ctx)
```

Reply:
So the g.wait will wait until so tasks are done before returning the first error. The behavior am trying to accomplish here is when the first routing errors or ask tasks exit and the error returns immediately

Reply:
I was so focused on typing out the test I forgot what the actual code should be… one sec I’ll fix this up

Reply:
hopefully this is better now